event.go

```diff
@@ -46,6 +49,32 @@ func NewEvent(t EventType, o Identifiable) *Event {
 	return NewEventWithEncoding(t, o, EncodingTypeMSGPACK)
 }
 
+// NewErrorEvent returns a new (error) Event embedded with the provided elemental.Error
+func NewErrorEvent(ee Error, encoding EncodingType) *Event {
+
+	data, err := Encode(encoding, ee)
+	if err != nil {
+		panic(fmt.Sprintf("unable to create new error event: %s", err))
+	}
+
+	event := &Event{
+		Type: EventError,
+		// TODO: should we even populate this field at all? is it sufficient that the client can assume they should just
+		// deserialize this into an elemental error since the event type is error?
+		Identity:  string(EventError),
+		Timestamp: time.Now(),
+		Encoding:  encoding,
+	}
+
+	if encoding == EncodingTypeJSON {
```

Comment:
can you extract a function that takes the *event and sets the correct data and reuse it here and in NewEventWithEncoding. I don't want this code to have any chance to diverge by mistake

Reply:
Donezo ✅